packages/server/src/tests/header/headerSelection.ts

```diff
@@ -107,91 +91,58 @@ const supportersTest: HeaderTest = {
-const usEoyPeriodEnd = new Date(2022, 1, 1);
+    const { showSupportMessaging, countryCode } = targeting;
 
-const isInUsEoyPeriod = (date: Date): boolean => {
-    return isAfter(date, usEoyPeriodStart) && isBefore(date, usEoyPeriodEnd);
-};
+    const countryGroupId = countryCodeToCountryGroupId(countryCode.toUpperCase());
+
+    const selectedTest: HeaderTest | undefined = allTests.find(test => {
+        const { isOn, userCohort, locations } = test;
+
+        if (!isOn) {
+            return false;
+        }
+
+        if (showSupportMessaging && !isNotReportedAsSupporter.includes(userCohort)) {
+            return false;
+        }
+
+        if (!locations.includes(countryGroupId)) {
+            return false;
+        }
 
-const hasNotContributedDuringUsEoyPeriod = (lastOneOffContributionDate?: string): boolean => {
-    if (!lastOneOffContributionDate) {
         return true;
+    });
+
+    // Failed to find a matching test, or the matching test has an empty variants Array
+    if (!selectedTest || !selectedTest.variants.length) {
+        return null;
     }
 
-    return isBefore(new Date(Date.parse(lastOneOffContributionDate)), usEoyPeriodStart);
-};
+    const selectedVariant: HeaderVariant = selectVariant(selectedTest, targeting.mvtId);
 
-const getNonSupportersTest = (edition: Edition): HeaderTest => {
-    if (edition === 'UK') {
-        return nonSupportersTestUK;
-    }
-    if (edition === 'US' && isInUsEoyPeriod(new Date())) {
-        return nonSupportersTestUS();
-    }
-    return nonSupportersTestNonUK;
-};
+    selectedVariant.modulePathBuilder = modulePathBuilder;
 
-const getSupportersTest = (edition: Edition, lastOneOffContributionDate?: string): HeaderTest => {
-    if (
-        edition === 'US' &&
-        isInUsEoyPeriod(new Date()) &&
-        hasNotContributedDuringUsEoyPeriod(lastOneOffContributionDate)
-    ) {
-        return supportersTestUS;
-    }
-    return supportersTest;
+    return {
+        test: selectedTest,
+        variant: selectedVariant,
+        moduleName: header.name,
+        modulePathBuilder,
+    };
 };
 
 export const selectHeaderTest = (
     targeting: HeaderTargeting,
 ): Promise<HeaderTestSelection | null> => {
-    const select = (): HeaderTest => {
-        if (targeting.showSupportMessaging) {
-            return getNonSupportersTest(targeting.edition);
-        } else {
-            return getSupportersTest(targeting.edition, targeting.lastOneOffContributionDate);
-        }
-    };
-
-    const test = select();
-    const variant = test.variants[targeting.mvtId % test.variants.length];
-    if (test && variant) {
-        return Promise.resolve({
-            test,
-            variant,
-            moduleName: header.name,
-            modulePathBuilder: variant.modulePathBuilder,
-        });
-    }
-    return Promise.resolve(null);
+    return fetchConfiguredHeaderTestsCached()
+        .then((allTests: HeaderTest[]) => selectBestTest(targeting, allTests))
```

Comment:
`allTests` doesn't include hardcoded tests, so perhaps a misleading name?
Though I think it would be nicer to include any hardcoded tests in the array passed in here

Reply:
Agreed, and done. Jest test updated.